skbio/stats/composition.py

```diff
@@ -608,10 +610,272 @@ def centralize(mat):
+    by passing in a different statistical function.
+
+
+    Parameters
+    ----------
+    mat: pd.DataFrame or array_like
+       A 2D matrix where
+       rows = samples
+       columns = features
+    cat: pd.Series or array_like
+       Vector of categories
+    alpha : float
+       Significance level for each of the statistical tests
+    multicorr: bool
+       Runs multiple comparisons correction or not.
+       If specified, this will run Holm-Boniferroni
+       correction
+    tau: float
+       A constant used to determine an appropriate
+       cutoff (default=0.02)
+    theta : float
+       Lower bound for the proportion of differiential features
+       (default=0.1)
+    func: function
+       A statistical signficance function to test for
+       signficance between classes.
+       The default is scipy.stats.ttest_ind
+
+    Returns:
+    --------
+    W : pd.Series
+        List of W statistics
+    reject : pd.Series
+        Indicates if the null hypothesis has been rejected
+
+    References
+    ----------
+    ..[1] S. Mandal, 'Analysis of composition of microbiomes:
+          a novel method for studying microbial composition'
+
+    Examples
+    --------
+    First import all of the necessary modules
+
+    >>> from skbio.stats.composition import ancom
+    >>> import pandas as pd
+
+    Now lets load in a pandas dataframe with sample and feature ids
+    for our data matrix.
+    >>> table = pd.DataFrame(
+    ...     [[10., 11., 10., 10., 10., 10., 10.],
+    ...      [10.5, 11.5, 10.5, 10.5, 10.5, 10.5, 10.5],
+    ...      [10., 11., 10., 10., 10., 10., 10.],
+    ...      [20., 21., 10., 10., 10., 10., 10.],
+    ...      [20.5, 21.5, 10.5, 10.5, 10.5, 10.5, 10.5],
+    ...      [20.3, 21.3, 10.2, 10.3, 10.1, 10.6, 10.4]],
+    ...     index=['s1','s2','s3','s4','s5','s6'],
+    ...     columns=['b1','b2','b3','b4','b5','b6','b7'])
+
+    Then create a create a category vector.  In this scenerio, there
+    are only two class, so the first three samples fall under the first
+    class while the last three samples fall under the last class
+    >>> cats = pd.Series([0, 0, 0, 1, 1, 1],
+    ...                  index=['s1','s2','s3','s4','s5','s6'])
+
+    Now run ancom and see if there are any features that have any
+    significant differences
+    >>> W, reject = ancom(table, cats)
+    >>> print(W)
+    b1    6
+    b2    6
+    b3    2
+    b4    2
+    b5    2
+    b6    2
+    b7    2
+    dtype: float64
+    >>> print(reject)
+    b1     True
+    b2     True
+    b3    False
+    b4    False
+    b5    False
+    b6    False
+    b7    False
+    dtype: bool
+    """
+    if len(mat) != len(cats):
+        raise ValueError('The number of samples in mat needs'
+                         'to be the same as the number of samples'
+                         'in cats')
+    mat = _check_composition(mat, ignore_zeros=False)
+    cats = pd.Series(cats)
+
+    mat = mat.sort_index()
```

Comment:
Ah, I see, then it makes sense to keep as is.

On (Oct-12-15| 9:14), mortonjt wrote:
> > +    b2     True
> > +    b3    False
> > +    b4    False
> > +    b5    False
> > +    b6    False
> > +    b7    False
> > +    dtype: bool
> > +    """
> > +    if len(mat) != len(cats):
> > +        raise ValueError('The number of samples in mat needs'
> > +                         'to be the same as the number of samples'
> > +                         'in cats')
> > +    mat = _check_composition(mat, ignore_zeros=False)
> > +    cats = pd.Series(cats)
> > +
> > +    mat = mat.sort_index()
> 
> Good point for `pd.DataFrame`.  Doesn't look like this functionality is available for `pd.Series` as of `pandas==0.16`
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/scikit-bio/pull/1114/files#r41773146